cpp/oneapi/dal/algo/kmeans_init/backend/cpu/compute_kernel_dense.cpp

```diff
@@ -43,9 +45,18 @@ static compute_result<Task> call_daal_kernel(const context_cpu& ctx,
     const std::int64_t column_count = data.get_column_count();
     const std::int64_t cluster_count = desc.get_cluster_count();
 
+    //number of trials to pick each centroid from, 2 + int(ln(cluster_count)) works better than vanilla kmeans++
+    //https://github.com/scikit-learn/scikit-learn/blob/a63b021310ba13ea39ad3555f550d8aeec3002c5/sklearn/cluster/_kmeans.py#L108
+    std::int64_t trial_count = desc.get_local_trials_count();
+    if (trial_count == -1) {
+        const auto additional = std::log(cluster_count);
+        trial_count = 2 + std::int64_t(additional);
+    }
+
```

Comment:
What kind of improvements this give? It seems this changes original behavior. Please also reflect it in the documentation if not already done.

Reply:
This is actually an original behavior in daal and oneDAL distributed, somehow was missed on oneDAL CPU